<?xml version="1.0"?>
<launch>
	<arg name="robot" default="nusim" doc="the hostname of the turtlebot, localhost for simulation"/>
	
	<param name="unknown_assoc" value="1"/>
	<param name="dst_thresh" value="0.4"/>  
	<param name="dst_metric" value="2"/> 
	<param name="cluster_thresh" value="0.3"/>
	<param name="min_cluster_num" value="5"/>
	<param name="min_angle" value="80"/>
	<param name="max_angle" value="160"/>
	<param name="angle_std" value="0.3"/>
	<param name="min_circle_rad" value="0.04"/>
	<param name="max_circle_rad" value="0.10"/>
		
	<group if="$(eval robot != 'nusim')">
		<machine name="turtlebot" address="localhost" env-loader="/home/ubuntu/loader.sh"/>
		<include file="$(find nuslam)/launch/slam.launch">
			<arg name="robot" value="localhost"/>
		</include>
	</group>
	
	<group if="$(eval robot == 'nusim')">
		<include file="$(find nuslam)/launch/slam.launch">
			<arg name="robot" value="$(arg robot)"/>
			<arg name="use_rviz" value="false"/>
		</include>
		<node pkg="tf2_ros" type="static_transform_publisher" name="link2_broadcaster" args="0 0 0 0 0 0 1 world map" />
		</group>
	
	<node pkg="nuslam" type="landmarks" name="landmarks" if = "$(eval robot == 'nusim')"/>
	<node machine="turtlebot" pkg="nuslam" type="landmarks" name="landmarks" if = "$(eval robot != 'nusim')"/>
</launch>